lazy_tensor_core/lazy_tensor_core/csrc/init_python_bindings.cpp

```diff
@@ -683,6 +683,63 @@ void InitLtcModuleBindings(py::module m) {
   m.def("_ltc_set_noop_execution_mode", [](bool enable_noop) {
     torch::lazy::LazyGraphExecutor::Get()->SetNoOpExecutionMode(enable_noop);
   });
+  m.def("_jit_function_call", [](py::args args, py::kwargs kwargs) {
+    HANDLE_TH_ERRORS
+
+    auto strongPtr = py::cast<torch::jit::StrongFunctionPtr>(args[0]);
+    torch::jit::Function& callee = *strongPtr.function_;
+
+    std::cerr << "function pointer = " << &callee << std::endl;
+
+    std::vector<c10::IValue> stack;
+    size_t tensor_args = 0;
+
+    std::cerr << "constructing arguments\n";
+    torch::lazy::BackendDevice device;
+    std::vector<torch::lazy::Value> tvals;
+
+    std::vector<c10::IValue> ivals;
+
+    // replace lazy tensors with meta tensors
+    // so we can do shape inference
+    for (auto i: c10::irange(args.size() - 1)) {
+      if (THPVariable_Check(args[i + 1].ptr())) {
+        auto& t = THPVariable_Unpack(args[i + 1].ptr());
+        auto lt = torch::lazy::GetLtcTensor(t);
+        device = lt.GetDevice();
+        tvals.push_back(lt.GetIrValue());
+        auto mt = at::empty_strided(t.sizes(), t.strides(), at::TensorOptions(at::kMeta).dtype(t.scalar_type()));
```

Comment:
can't just do args[i + 1].to(kMeta)?

Reply:
I could check; my concern was this could trigger materialization?

Reply:
actually scratch the .to suggestion, appears even that is not needed.

Our LazyNativeFunctions code just does this:

            auto out_meta = at::meta::nll_loss_backward(grad_output, self, target, weight, reduction, ignore_index, total_weight);
            
            where grad_output is the at::tensor that is passed into the nativefunction.  I guess you can just pass the lazytensor directly to meta func and it converts it inside?